utils/camera.js

```diff
@@ -9,41 +22,57 @@ module.exports = {
     }
 
     const shakeTimer = game.time.create(false);
+    const shakeInterval = options.randomizeInterval? 
+      Math.random() * options.shakeInterval + options.shakeInterval : 
+      options.shakeInterval;
 
-    const shakeRange = 20;
-    const shakeInterval = 60;
-
-    let shakeCount = 10;
+    shakeTimer.loop(options.shakeInterval, () => {
+      if (options.shakeCount === 0) {
+        // if end shake reset camera, stop shake timer and call callback
+        if (target) {
+          camera.follow(target);
+        }
 
-    shakeTimer.loop(
-      shakeInterval,
-      function doShake() {
-        if (!shakeCount) {
-          if (target) {
-            camera.follow(target);
-          }
+        if (typeof callback === 'function') {
+          callback();
+        }
 
-          shakeTimer.stop();
+        shakeTimer.stop();
+        return;
+      }
 
-          if (typeof callback === 'function') {
-            callback();
-          }
+      // Calculate camera shift
+      let shift1, shift2;
+      const shakeRangeHalved = options.shakeRange / 2;
 
-          return;
+      if (options.randomShake) {
+        shift1 = game.rnd.integerInRange(-shakeRangeHalved, shakeRangeHalved);
+        shift2 = game.rnd.integerInRange(-shakeRangeHalved, shakeRangeHalved);
+      } else {
+        if (options.shakeCount % 2){
+          shift1 = shift2 = -shakeRangeHalved;
+        } else{
+          shift1 = shift2 = shakeRangeHalved;
         }
+      }
 
-        let shift = shakeRange / 2;
-
-        if (shakeCount % 2) {
-          shift *= -1;
-        }
+      // Shake camera
+      if (options.shakeAxis === "y") {
+        game.camera.y += shift2;
+      }
 
-        camera.x += shift;
+      if (options.shakeAxis === "x") {
+        game.camera.x += shift1;
+      }
 
-        shakeCount--;
+      if (options.shakeAxis === "xy") {
+        game.camera.x += shift1;
+        game.camera.y += shift2;
       }
```

Comment:
So we shake across the tl-br diagonal always if we aren't randomizing? I guess that's fine.